Tensile/Source/lib/CMakeLists.txt

```diff
@@ -68,9 +69,19 @@ add_library (TensileHost STATIC ${tensile_sources})
 
 target_include_directories(TensileHost INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/include")
 
+set(MSGPACK_DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/msgpack")
+include(ExternalProject)
+ExternalProject_Add(msgpack
+  GIT_REPOSITORY    https://github.com/msgpack/msgpack-c.git
+  GIT_TAG           cpp-3.2.1
+  CMAKE_ARGS        -DMSGPACK_CXX11=ON -DMSGPACK_BUILD_TESTS=OFF MSGPACK_BUILD_EXAMPLES=OFF -DCMAKE_INSTALL_PREFIX=${MSGPACK_DESTINATION}
+)
```

Comment:
Dont build msgpack in the cmake project. This should use the preinstalled version and should be able to find it with `find_package(msgpack-c)` and used with `target_link_libraries(TensileHost mspackc)`.

Also, ExternalProject is problematic as it can loose toolchain settings and you can't get the usage requirements from it. It should only be used for superprojects(which TensileHost is a project not a superproject).

Reply:
As of fda2250, we will be downloading an updated version of msgpack as the one included with the OS is very outdated.

Reply:
@pfultz2 is this new solution acceptable?

Reply:
Yes that is acceptable, we will install a newer version of msgpack with cget for miopentensile.